app/graphql/mutations/update_covid_run_mutation.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+module Mutations
+  # Mutation to update a COVID run.
+  class UpdateCovidRunMutation < BaseMutation
+    argument :id, ID, 'The ID of the Ont Run to update.', required: true
+    argument :properties, Types::Inputs::Ont::RunInputType, 'The properties to update on the run.',
+             required: true
+
+    field :run, Types::Outputs::Ont::RunType, 'The updated Run, or null if errors were thrown.',
+          null: true
+    field :errors, [String], 'An array of error messages thrown while updating the run.',
+          null: false
+
+    def resolve(id:, properties:)
+      run = Ont::Run.find_by(id: id)
+      return { run: nil, errors: ["Couldn't find the specified run."] } if run.nil?
+
+      update_run_state(run, properties.state)
+      errors = update_run_flowcells(run, properties.flowcells)
+
+      if !errors.nil? && errors.any?
+        { run: nil, errors: errors }
+      else
+        { run: run, errors: [] }
+      end
+    end
+
+    private
+
+    def update_run_state(run, state)
+      run.update(state: state) unless state.nil?
+    end
+
+    def update_run_flowcells(run, flowcell_specs)
+      return if flowcell_specs.nil? || flowcell_specs.count == 0
```

Comment:
I guess you'll be disappointed.  I'm happy to change this if we think that behaviour would be more desirable.  In reality, I don't think this will happen, otherwise you'll be left with runs that have no flowcells and are therefore invalid for submission. There would be nothing to submit a message for the run.

Reply:
But can I not create a run with no flowcells using the `CreateCovidRunMutation`? It seems contradictory that you can create a run with no flowcells, but not update a run to have no flowcells

Reply:
It may be that the creation workflow needs changing to not allow creating an empty run

Reply:
I'll make it generate an error if you pass in an empty array.  Only accepting a populated array or nil